src/textinput/TextInput.ts

```diff
@@ -0,0 +1,37 @@
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
+import { FormLabelMixin, FormLabelMixinProperties } from '@dojo/widget-core/mixins/FormLabel';
+import { v } from '@dojo/widget-core/d';
+import * as css from './styles/textinput.css';
+
+/**
+ * @type TextInputProperties
+ *
+ * Properties that can be set on a TextInput component
+ *
+ * @property type			Input type, e.g. text, email, tel, etc.
+ * @property onInput	Called when the node's 'input' event is fired
+ */
+export interface TextInputProperties extends ThemeableProperties, FormLabelMixinProperties {
+	type?: string;
+	onInput?(event: Event): void;
+}
+
+@theme(css)
+export default class TextInput extends ThemeableMixin(FormLabelMixin(WidgetBase))<TextInputProperties> {
+	onInput(event: Event) {
+		this.properties.onInput && this.properties.onInput(event);
+	}
+
+	render() {
```

Comment:
Thanks for the great examples @bitpshr. Having read all the facts and doing some more research I agree with your approach. I guess we would suggesting using `onchange` / `oninput` depending on the use case. I would imagine that this will be crucial for how we implement forms as the form inputs will need to be controlled as such.